Extensions/dist/themes.js

```diff
@@ -0,0 +1,99 @@
+/* jshint node:true */
+'use strict';
+
+var fs = require('fs'),
+	path = require('path');
+
+var resourceBuilder = require('./resources');
+
+/**
+ * Build the themes in `themePath`, writing the distribution json file
+ * into `distributionPath`.
+ * @param {String} themePath
+ * @param {String} distributionPath
+ * @exported
+ */
+function build(themePath, distributionPath) {
+	themeBuilder
+		.create(themePath, distributionPath)
+		.processThemes();
+}
+exports.build = build;
+
+/**
+ * @constructor
+ * @param {String} themePath Path from which to read theme files
+ * @param {String} distributionPath Path in which to write theme file
+ */
+var themeBuilder = Object.create(resourceBuilder);
+themeBuilder.create = function(resourcePath, distributionPath) {
+	return resourceBuilder.create.call(this, resourcePath, distributionPath);
+};
+
+/**
+ * Process all themes found in `themePath`
+ */
+themeBuilder.processThemes = function() {
+	var themeFiles = fs.readdirSync(this.resourcePath);
+	var themes = [];
+
+	themeFiles.forEach(function(fileName) {
+		// Look only for css files
+		var fileNameParts = fileName.split('.');
+		if (fileNameParts.length !== 2) {
```

Comment:
gulp build:themes is generating an empty themes.json file and it looks like this is the culprit. I believe that because each theme is in its own directory this is incorrect in assuming that there will be a css file.